File: Crio/StackAndQueue/6_KthMultiple/KthMultiple.py
def kthMultiple(k):
    # 1st element will be 1
    ans = 1
    
    # Make individual queue for 3, 5 and 7
    q3 = [] 
    q5 = [] 
    q7 = []
    
    for i in range(k-1):
        # Add next possible numbers to queue
        q3.append(ans * 3)
        q5.append(ans * 5)
        q7.append(ans * 7)

        # Find current number
        ans = min(q3[0], q5[0], q7[0])

        # Delete current number from all the queues
        if (ans == q3[0]):
            q3.pop(0)

        if (ans == q5[0]):
            q5.pop(0)

        if (ans == q7[0]):
            q7.pop(0)

    return ans

if __name__ == '__main__':
    k = int(input())
    result = kthMultiple(k)
    print(result)

File: Crio/StackAndQueue/6_KthMultiple/KthMultiple.cpp
#include <bits/stdc++.h>
using namespace std;

long long kthMultiple(long long k) {
    // CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
    long long a = 3, b = 5, c = 7, x = 1;
    queue<long long>q3, q5, q7;
    long long inf = 10000000000000000;
    for (long long i = 1; i < k; i++)
    {
        q3.push(x * 3);
        q5.push(x * 5);
        q7.push(x * 7);
        x = min(q3.front(), min(q5.front(), q7.front()));

        if (x == q3.front())
        {
            q3.pop();
        }
        if (x == q5.front())
        {
            q5.pop();
        }
        if (x == q7.front())
        {
            q7.pop();
        }
    }
    return x;
    // CRIO_SOLUTION_END_MODULE_L1_PROBLEMS
}

int main()
{
    long long k = 10;
    cin >> k;
    long long x = kthMultiple(k);
    cout << x << endl;
    return 0;
}